library/extractor/src/main/java/com/google/android/exoplayer2/extractor/VorbisUtil.java

```diff
@@ -248,6 +263,89 @@ public static CommentHeader readVorbisCommentHeader(
     return new CommentHeader(vendor, comments, length);
   }
 
+  /**
+   * Builds a metadata instance from Vorbis comments.
+   *
+   * METADATA_BLOCK_PICTURE comments will be transformed into picture frames.
+   * All others will be transformed into vorbis comments.
+   *
+   * @param vorbisComments The raw input of comments, as a key-value pair KEY=VAL.
+   * @return A metadata instance containing the processed tags.
+   */
+  @Nullable
+  public static Metadata buildMetadata(List<String> vorbisComments) {
+    return buildMetadata(vorbisComments, Collections.emptyList());
+  }
+
+  /**
+   * Builds a metadata instance from Vorbis comments.
+   *
+   * METADATA_BLOCK_PICTURE comments will be transformed into picture frames.
+   * All others will be transformed into vorbis comments.
+   *
+   * @param vorbisComments The raw input of comments, as a key-value pair KEY=VAL.
+   * @param pictureFrames Any picture frames that were parsed beforehand.
+   * @return A metadata instance containing the processed tags.
+   */
+  @Nullable
+  public static Metadata buildMetadata(List<String> vorbisComments, List<PictureFrame> pictureFrames) {
+    if (vorbisComments.isEmpty() && pictureFrames.isEmpty()) {
+      return null;
+    }
+
+    ArrayList<Metadata.Entry> metadataEntries = new ArrayList<>();
+    for (int i = 0; i < vorbisComments.size(); i++) {
+      String vorbisComment = vorbisComments.get(i);
+      String[] keyAndValue = Util.splitAtFirst(vorbisComment, SEPARATOR);
+
+      if (keyAndValue.length != 2) {
+        Log.w(TAG, "Failed to parse Vorbis comment: " + vorbisComment);
+      } else {
+        if (keyAndValue[0].equals("METADATA_BLOCK_PICTURE")) {
+          // This tag is a special cover art tag, outlined by
+          // https://wiki.xiph.org/index.php/VorbisComment#Cover_art.
+          // Decode it from Base64 and transform it into a PictureFrame.
+          try {
+            byte[] decoded = Base64.decode(keyAndValue[1], Base64.DEFAULT);
+            metadataEntries.add(buildPictureFrame(new ParsableByteArray(decoded)));
+          } catch (Exception e) {
+            Log.w(TAG, "Failed to parse vorbis picture");
+          }
+        } else {
+          VorbisComment entry = new VorbisComment(keyAndValue[0], keyAndValue[1]);
+          metadataEntries.add(entry);
+        }
+      }
+    }
+    metadataEntries.addAll(pictureFrames);
+
+    return metadataEntries.isEmpty() ? null : new Metadata(metadataEntries);
+  }
+
+  /**
+   * Transforms a picture metadata block into a picture frame.
+   *
+   * @param scratch The bytes of the picture.
+   * @return A picture frame from the scratch data.
+   */
+  static PictureFrame buildPictureFrame(ParsableByteArray scratch) {
```

Comment:
It can't be private as `FlacExtractor` needs to call it to it's picture blocks:

```java
ParsableByteArray pic_scratch = new ParsableByteArray(length);
input.readFully(pic_scratch.getData(), 0, length);
pic_scratch.skipBytes(FlacConstants.METADATA_BLOCK_HEADER_SIZE);
PictureFrame pictureFrame = VorbisUtil.buildPictureFrame(pic_scratch);
```

Perhaps we should make it package-private?

Reply:
Resolved this by making it package-private in 7a88829.